Discuss auth-related changes in CHANGELOG.md (#1154)
Azure/azure-dev: cli/azd/CHANGELOG.md

## Upcoming (Unreleased)

> ### Important Notes

Reviewer: IMHO, this should be part of `breaking changes`.
the Breaking changes are the most important notes from the changelog.  

By having `Important Notes`, I would wonder:
- Is it a breaking change?  is it a new feature?  are other sections not important notes? :)
- Are we deviating from the standard Azure SDK's release changelog sections?  
- Usually, within the Azure SDK, there is a blog post which highlights the most important changes on the top, by just elevating one of the entries of the changelog.  Should we have a note like the one in this PR to be on the top of some of the @savannahostrowski 's blog posts instead? (while keeping a single entry here within the breaking changes?)

Reviewer 2: We will include this update in the January release post. @ellismg is there an issue associated with this PR? Looking to tag it as "newsworthy" so we don't forget to highlight it in the next monthly update.

Author: @vhvb1989 I had moved this up to the "top" of the release notes because I wanted it to the first thing people saw.  I actually still have a note about this in the breaking changes section (if you expand the context of the diff, I think you'll see it, but here's the content):

- [[#1022]](https://github.com/Azure/azure-dev/pull/1022) `azd` no longer uses the `az` CLI to authenticate with Azure by default. You will need to run `azd login` after upgrading. You may run `azd config set auth.useAzCliAuth true` to restore the old behavior of using `az` for authentication.

My general thinking here was that we'd use the top of the changelog to really highlight things, and then have the list of changes come later, but I see your point about stuff being regular being useful I'll move the prose down into that section.

As part of this release, `azd` has been updated to no longer depend on `az`. As part of these changes, you must now explicitly log into `azd` instead of logging into `az`. After upgrading to this version of `azd`, please run `azd login` to log in.

> If you have existing pipelines that use `azd` you will also need to update your pipelines which use `azd` to make sure it can authenticate to Azure.

Reviewer 3: ```suggestion
If you have existing pipelines that use `azd`, you will need to update your pipelines to use the new `azd` login methods when authenticating against Azure.
```

> For Pipelines which use GitHub Actions:

Reviewer 3: ```suggestion
For GitHub Actions pipelines:
```

> Update your `azure-dev.yml` to stop using the `azure/login@v1` action and log in using `azd` directly. To do so, replace:

Reviewer 3: ```suggestion
Update your `azure-dev.yml` to stop using the `azure/login@v1` action, and instead log in using `azd` directly. To do so, replace:
```

```yaml
      - name: Log in with Azure
        uses: azure/login@v1
        with:
          creds: ${{ secrets.AZURE_CREDENTIALS }}
```

with

```yaml
      - name: Log in with Azure
        run: |
          $info = $Env:AZURE_CREDENTIALS | ConvertFrom-Json -AsHashtable;
          Write-Host "::add-mask::$($info.clientSecret)"

          azd login `
            --client-id "$($info.clientId)" `
            --client-secret "$($info.clientSecret)" `
            --tenant-id "$($info.tenantId)"
        shell: pwsh
        env:
          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
```

> For Pipelines which use Azure Dev Ops:

Reviewer 3: ```suggestion
For Azure DevOps pipelines:
```

> Update your `azure-dev.yml` file to force `azd` to use the `az` for authentication.  To do so, add a new step before any other steps which use `azd`:

Reviewer 3: ```suggestion
Update your `azure-dev.yml` file to force `azd` to use `az` for authentication.  To do so, add a new step before any other steps which use `azd`:
```

```yaml
      - name: Configure azd to use az for authentication
        run: |

>           azd config set auth.useAzCliAuth "true"

Reviewer 3: Does this need to be `script` and `displayName`?

```yaml
- script: |
    azd config set auth.useAzCliAuth "true"
   displayName: Configure azd to use az for authentication    
```

https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/cross-platform-scripting?view=azure-devops&tabs=yaml

Our pipelines: https://github.com/Azure/azure-dev/blob/5bee2119474d9c2b6842b0307557ac41cfd9fb41/templates/common/.azdo/pipelines/terraform/azure-dev.yml#L11

Author: You are correct (and I was correct when updating the pipelines in the past) - I was looking at the wrong change when I moved this content into the CHANGELOG.
